src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -133,8 +133,18 @@ public string TestCode
         /// </summary>
         public List<string> DisabledDiagnostics { get; } = new List<string>();
 
+        /// <summary>
+        /// Gets or sets the reference assemblies to use.
+        /// </summary>
         public ReferenceAssemblies ReferenceAssemblies { get; set; } = ReferenceAssemblies.Default;
 
+        /// <summary>
+        /// Gets or sets an additional verifier for a diagnostic.
+        /// The action compares actual <see cref="Diagnostic"/> and the expected
+        /// <see cref="DiagnosticResult"/> based on custom test requirements not yet supported by the test framework.
+        /// </summary>
+        public Action<ImmutableArray<DiagnosticAnalyzer>, Diagnostic, DiagnosticResult, IVerifier>? DiagnosticVerifier { get; set; }
```

Comment:
Can we remove the analyzer portion?

```suggestion
        public Action<Diagnostic, DiagnosticResult, IVerifier>? DiagnosticVerifier { get; set; }
```

💭 I'm tempted to go as far as removing the `DiagnosticResult` portion as well.

Reply:
That means callers cant use the following method for consistency:
```
     protected string FormatVerifierMessage(ImmutableArray<DiagnosticAnalyzer> analyzers, Diagnostic actual, DiagnosticResult expected, string message)
```
Is that ok?

Reply:
re: `DiagnosticResult` for `expected`.  I'd like to keep it so that it can be a key to a dictionary containing expected data.

Reply:
> That means implementors can't use the following method for consistency

The helper wouldn't produce an actionable error message for this case, so I'm not sure how much it helps.

> I'd like to keep it so that it can be a key to a dictionary containing expected data.

📝 This is already not supported. `DiagnosticResult` is a value type which is not equatable.

Let me think about this a bit. If I can't find an "obviously better" option then the one here seems reasonable.

Reply:
I do use `Span`s as keys / comparisons in some of my analyzers to determine if I'm matching up the same piece of code.

Reply:
@sharwell are we okay to go ahead with Niko's contribution?

Reply:
I am